Prevent normalization of polymorphic association

eloyesp opened the pull request:
Polymorphic associations have no klass names, so those should be not
normalized. Fixes #588.

- [x] Add tests
- [x] Changelog entry

I might need help with the tests.

eloyesp commented:
@coorasse any thought about this pull-request?

coorasse commented:
Hi! Thanks for your PR. 
I like your solution, since this case is currently not well managed IMHO. 
In an optimal case, it should be normalised to 
```
expect(rule.conditions).to eq(attachments: { record_id: 11, record_type: 'Article' })
```
I'd merge it by adding a TODO that explains the best scenario, not currently possible.
What are your thought about it?

eloyesp commented:
Let me check if I can get it working that way.

eloyesp commented:
@coorasse The problem with that normalization is that it depends on querying for id, it will not work on other fields.

``` ruby
    # query for id
    rule = CanCan::Rule.new(true, :read, Blob, articles: { id: 11 })
    CanCan::ModelAdapters::ConditionsNormalizer.normalize(Blob, [rule])
    expect(rule.conditions).to eq(attachments: { record_id: 11, record_type: 'Article' })

    # but does not work for other queries
    rule = CanCan::Rule.new(true, :read, Blob, articles: { status: 'draft' })
    # expect(rule.conditions).to eq(attachments: { ???? })
```

I've fixed some missing options on the sample associations and updated for rails 6.0 rc2.

Regards.

coorasse commented:
It could be 
```expect(rule.conditions).to eq(attachments: { record: {id: 11, type: 'Article', status: 'draft' })```?
We need to rebase, also. thanks

eloyesp commented:
@coorasse I've done the rebase.

> It could be
>     expect(rule.conditions).to eq(attachments: { record: {id: 11, type: 'Article', status: 'draft' })?

I'm not sure if that would work in Rails, but the main issue is that I'm not sure how to modify the actual code to do it.
